### PR TITLE
GH-245: Add Orientation Sensor

### PR DESCRIPTION
**Pull request by charlespetzold**
# Description of Change

Add Orientation Sensor to API (rationale and description in issue #245)

# API Changes

```
public static class OrientationSensor
{
    public static event OrientationSensorChangedEventHandler ReadingChanged;
    public static bool IsMonitoring { get; private set; }
    public static void Start(SensorSpeed sensorSpeed);
    public static void Stop();
}

public delegate void OrientationSensorChangedEventHandler(OrientationSensorChangedEventArgs e);

public class OrientationSensorChangedEventArgs : EventArgs
{
    public OrientationSensorData Reading { get; }
}

public struct OrientationSensorData
{
    public System.Numerics.Quaternion Orientation { get; }
}
```

### CODE REVIEW

**Comment by dend** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Samples/Samples/Samples.csproj](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/Samples.csproj) | :white_check_mark:Succeeded | |
[Samples/Samples/View/AllSensorsPage.xaml](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/View/AllSensorsPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/View/OrientationSensorPage.xaml](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/View/OrientationSensorPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/View/OrientationSensorPage.xaml.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/View/OrientationSensorPage.xaml.cs) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/HomeViewModel.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/ViewModel/HomeViewModel.cs) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/OrientationSensorViewModel.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/ViewModel/OrientationSensorViewModel.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/OrientationSensor/OrientationSensor.android.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Xamarin.Essentials/OrientationSensor/OrientationSensor.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/OrientationSensor/OrientationSensor.ios.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Xamarin.Essentials/OrientationSensor/OrientationSensor.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/OrientationSensor/OrientationSensor.netstandard.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Xamarin.Essentials/OrientationSensor/OrientationSensor.netstandard.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/OrientationSensor/OrientationSensor.shared.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Xamarin.Essentials/OrientationSensor/OrientationSensor.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/OrientationSensor/OrientationSensor.uwp.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Xamarin.Essentials/OrientationSensor/OrientationSensor.uwp.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/Xamarin.Essentials.csproj](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Xamarin.Essentials/Xamarin.Essentials.csproj) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C5%5C12%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201805121614181873-249%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=tpNHQ%2Bp%2F9T5WtLiOEx1L8yXlO2uisWc1H5672DRg4%2FQ%3D&st=2018-05-12T16%3A10%3A15Z&se=2018-06-12T16%3A15%3A15Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: Samples/Samples/Samples.csproj [hunk 1]
```diff
@@ -20,6 +20,12 @@
   <ItemGroup>
     <ProjectReference Include="..\..\Xamarin.Essentials\Xamarin.Essentials.csproj" />
   </ItemGroup>
+
+  <ItemGroup>
```
Comment (jamesmontemagno) [inline Samples/Samples/Samples.csproj]: This stuff gets auto added, but can be removed.

File: Xamarin.Essentials/Xamarin.Essentials.csproj [hunk 2]
```diff
@@ -73,6 +73,10 @@
     <Compile Include="**\*.ios.cs" />
     <Compile Include="**\*.ios.*.cs" />
   </ItemGroup>
+  <ItemGroup>
```
Comment (jamesmontemagno) [inline Xamarin.Essentials/Xamarin.Essentials.csproj]: Can also be removed.

**Comment by dend** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Samples/Samples/Samples.csproj](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/Samples.csproj) | :white_check_mark:Succeeded | |
[Samples/Samples/View/AllSensorsPage.xaml](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/View/AllSensorsPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/View/OrientationSensorPage.xaml](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/View/OrientationSensorPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/View/OrientationSensorPage.xaml.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/View/OrientationSensorPage.xaml.cs) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/HomeViewModel.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/ViewModel/HomeViewModel.cs) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/OrientationSensorViewModel.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/ViewModel/OrientationSensorViewModel.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/OrientationSensor/OrientationSensor.android.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Xamarin.Essentials/OrientationSensor/OrientationSensor.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/OrientationSensor/OrientationSensor.ios.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Xamarin.Essentials/OrientationSensor/OrientationSensor.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/OrientationSensor/OrientationSensor.netstandard.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Xamarin.Essentials/OrientationSensor/OrientationSensor.netstandard.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/OrientationSensor/OrientationSensor.shared.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Xamarin.Essentials/OrientationSensor/OrientationSensor.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/OrientationSensor/OrientationSensor.uwp.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Xamarin.Essentials/OrientationSensor/OrientationSensor.uwp.cs) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C5%5C12%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201805121647451067-249%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=gl7IQMyl8LJLtY9Szto1rbQg3b6LHTgO%2FGpevGDUgzA%3D&st=2018-05-12T16%3A43%3A41Z&se=2018-06-12T16%3A48%3A41Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by jamesmontemagno** [comment]
Build

**Comment by mattleibow** [comment]
Nice!

File: Samples/Samples/ViewModel/HomeViewModel.cs [hunk 3]
```diff
@@ -28,8 +28,8 @@ public HomeViewModel()
                     "📏",
                     "All Sensors",
                     typeof(AllSensorsPage),
-                    "Have a look at the accelerometer, compass, gyroscope and magnetometer.",
-                    new[] { "accelerometer", "compass", "gyroscope", "magnetometer", "sensors", "hardware", "device" }),
+                    "Have a look at the accelerometer, compass, gyroscope, magnetometer, and orientationsensor.",
+                    new[] { "accelerometer", "compass", "gyroscope", "magnetometer", "orientationsensor", "sensors", "hardware", "device" }),
```
Comment (mattleibow) [inline Samples/Samples/ViewModel/HomeViewModel.cs]: maybe just `"orientation"`

File: Samples/Samples/ViewModel/HomeViewModel.cs [hunk 4]
```diff
@@ -120,6 +120,12 @@ public HomeViewModel()
                     typeof(MagnetometerPage),
                     "Detect device's orientation relative to Earth's magnetic field.",
                     new[] { "compass", "magnetometer", "sensors", "hardware", "device" }),
+                new SampleItem(
+                    "📏",
+                    "OrientationSensor",
```
Comment (mattleibow) [inline Samples/Samples/ViewModel/HomeViewModel.cs]: Maybe `"Orientation Sensor"`

File: Samples/Samples/ViewModel/HomeViewModel.cs [hunk 5]
```diff
@@ -120,6 +120,12 @@ public HomeViewModel()
...
+                    typeof(OrientationSensorPage),
+                    "Retrieve orientation of the device in 3D space.",
+                    new[] { "orientationsensor", "sensors", "hardware", "device" }),
```
Comment (mattleibow) [inline Samples/Samples/ViewModel/HomeViewModel.cs]: `"orientation"`

**Comment by dend** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[Samples/Samples/Samples.csproj](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/Samples.csproj) | :white_check_mark:Succeeded | |
[Samples/Samples/View/AllSensorsPage.xaml](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/View/AllSensorsPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/View/OrientationSensorPage.xaml](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/View/OrientationSensorPage.xaml) | :white_check_mark:Succeeded | |
[Samples/Samples/View/OrientationSensorPage.xaml.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/View/OrientationSensorPage.xaml.cs) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/HomeViewModel.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/ViewModel/HomeViewModel.cs) | :white_check_mark:Succeeded | |
[Samples/Samples/ViewModel/OrientationSensorViewModel.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Samples/Samples/ViewModel/OrientationSensorViewModel.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/OrientationSensor/OrientationSensor.android.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Xamarin.Essentials/OrientationSensor/OrientationSensor.android.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/OrientationSensor/OrientationSensor.ios.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Xamarin.Essentials/OrientationSensor/OrientationSensor.ios.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/OrientationSensor/OrientationSensor.netstandard.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Xamarin.Essentials/OrientationSensor/OrientationSensor.netstandard.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/OrientationSensor/OrientationSensor.shared.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Xamarin.Essentials/OrientationSensor/OrientationSensor.shared.cs) | :white_check_mark:Succeeded | |
[Xamarin.Essentials/OrientationSensor/OrientationSensor.uwp.cs](https://github.com/charlespetzold/Essentials/blob/add-orientation-sensor/Xamarin.Essentials/OrientationSensor/OrientationSensor.uwp.cs) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C5%5C13%5Cf63aadd0-c8f5-13ca-04e7-037e45f8c0b9%5CPullRequest%5C201805131454444856-249%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=AGVokShxEgFAdx%2FuLSkKt5ntQ3KG14NURwGjY26qI9o%3D&st=2018-05-13T14%3A50%3A40Z&se=2018-06-13T14%3A55%3A40Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by mattleibow** [comment]
build

**Pull request merged**